cmd/wallet-web/src/components/DIDManagement/Preferences.vue

```diff
@@ -0,0 +1,151 @@
+      <label for="verification-method" class="mb-1 text-base">Key ID:</label>
+      <select
+        id="verification-method"
+        v-model="verificationMethod"
+        class="mb-5 w-full max-w-full border-b"
+        name="Verification Method"
+      >
+        <option value="default">Use Default</option>
+        <option v-for="keyId in keyIDs" :key="keyId" :value="keyId">
+          {{ keyId }}
+        </option>
+      </select>
+    </div>
+
+    <div class="flex flex-col flex-grow justify-start items-start w-full">
+      <label for="signature-type-selector" class="mb-1 text-base">Update Signature Type:</label>
+      <select
+        id="signature-type-selector"
+        v-model="selectedSignType"
+        class="mb-5 w-full max-w-full border-b"
+        name="Signature Type"
+      >
+        <option
+          v-for="signatureType in allSignatureTypes"
+          :key="signatureType.id"
+          :value="signatureType.id"
+        >
+          {{ signatureType.id }}
+        </option>
+      </select>
+    </div>
+
+    <button class="mt-5 btn-primary" :disabled="!preferencesChanged" @click="updatePreferences">
+      Update Preferences
+    </button>
+  </div>
+</template>
+
+<script>
+import { DIDManager, WalletUser } from '@trustbloc/wallet-sdk';
+import { mapActions, mapGetters } from 'vuex';
+import { getDIDVerificationMethod } from '@/pages/mixins';
+export default {
+  name: 'Preferences',
+  props: {
+    allDIDs: {
+      type: Array,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      keyID: '',
+      allSignatureTypes: [{ id: 'Ed25519Signature2018' }, { id: 'JsonWebSignature2020' }],
+      selectedDID: '',
+      selectedSignType: '',
+      verificationMethod: '',
+      keyIDs: [],
+      preference: {},
+    };
+  },
+  computed: {
+    preferencesChanged() {
+      const { controller, proofType, verificationMethod } = this.preference;
+      if (controller != this.selectedDID) {
+        return true;
+      }
+      if (proofType != this.selectedSignType) {
+        return true;
+      }
+      if (
+        verificationMethod != (this.verificationMethod === 'default' ? '' : this.verificationMethod)
+      ) {
+        return true;
+      }
+      return false;
+    },
+  },
+  created: function () {
+    const agent = this.getAgentInstance();
+    const { user } = this.getCurrentUser().profile;
+    this.walletUser = new WalletUser({ agent, user });
+    this.didManager = new DIDManager({ agent, user });
+    this.updateVerificationMethod();
+    this.listDIDs();
+  },
+  methods: {
+    ...mapGetters('agent', { getAgentInstance: 'getInstance' }),
+    ...mapGetters(['getCurrentUser']),
+    ...mapActions(['refreshUserPreference']),
+    listDIDs: async function () {
+      let { contents } = await this.didManager.getAllDIDs(this.getCurrentUser().profile.token);
+      const newAllDIDs = Object.keys(contents).map((k) => contents[k].DIDDocument);
+      this.$emit('update:allDIDs', newAllDIDs);
```

Comment:
what is the purpose of this event?

Reply:
To update the `allDIDs` array in `DIDManagement.vue`

Reply:
But all DIDs are being displayed in this template, you didn't delete `DIDManagement.vue`?

Reply:
May be I am confused,

preferences template: displays list of DIDs and gives selection option.
create orb: create orb DID and emits event to update list in preferences tab.
import did: imports DID and emits event to update list in preferences tab.

Reply:
That's correct, and the `DIDManagement.vue` component renders those three components that you mentioned in the tabbed interface. It also stores the `allDIDs` array, because it is logically shared between the three tabs

Reply:
I renamed this event more explicitly to avoid confusion. It is needed to emit the event which would update the `allDIDs` array in the `DIDManagement.vue` and thus, update the list of DIDs in the preferences tab.